src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIAnalysesOutputsService.java

```diff
@@ -0,0 +1,172 @@
+import ca.corefacility.bioinformatics.irida.ria.web.components.AnalysisOutputFileDownloadManager;
+import ca.corefacility.bioinformatics.irida.service.AnalysisSubmissionService;
+import ca.corefacility.bioinformatics.irida.service.user.UserService;
+import ca.corefacility.bioinformatics.irida.service.workflow.IridaWorkflowsService;
+
+import com.github.jsonldjava.shaded.com.google.common.base.Strings;
+
+/**
+ * UI Service for all things related to project single sample analysis outputs.
+ */
+@Component
+@Scope("session")
+public class UIAnalysesOutputsService {
+
+	private AnalysisSubmissionService analysisSubmissionService;
+	private IridaWorkflowsService workflowsService;
+	private UserService userService;
+	private AnalysisOutputFileDownloadManager analysisOutputFileDownloadManager;
+
+	@Autowired
+	public UIAnalysesOutputsService(AnalysisSubmissionService analysisSubmissionService,
+			IridaWorkflowsService workflowsService, UserService userService,
+			AnalysisOutputFileDownloadManager analysisOutputFileDownloadManager) {
+		this.analysisSubmissionService = analysisSubmissionService;
+		this.workflowsService = workflowsService;
+		this.userService = userService;
+		this.analysisOutputFileDownloadManager = analysisOutputFileDownloadManager;
+	}
+
+	/**
+	 * Get all the shared single sample analysis outputs for the project
+	 *
+	 * @param projectId {@link ca.corefacility.bioinformatics.irida.model.project.Project} id
+	 * @return a list of filtered {@link ProjectSampleAnalysisOutputInfo} single sample analysis outputs
+	 */
+	public List<ProjectSampleAnalysisOutputInfo> getSharedSingleSampleOutputs(Long projectId) {
+		List<ProjectSampleAnalysisOutputInfo> projectSampleAnalysisOutputInfos = analysisSubmissionService.getAllAnalysisOutputInfoSharedWithProject(
+				projectId);
+		return getSingleSampleAnalysisOutputsInfo(projectSampleAnalysisOutputInfos);
+	}
+
+	/**
+	 * Get all the automated single sample analysis outputs for the project
+	 *
+	 * @param projectId {@link ca.corefacility.bioinformatics.irida.model.project.Project} id
+	 * @return a list of filtered {@link ProjectSampleAnalysisOutputInfo} single sample analysis outputs
+	 */
+	public List<ProjectSampleAnalysisOutputInfo> getAutomatedSingleSampleOutputs(Long projectId) {
+		List<ProjectSampleAnalysisOutputInfo> projectSampleAnalysisOutputInfos = analysisSubmissionService.getAllAutomatedAnalysisOutputInfoForAProject(
+				projectId);
+		return getSingleSampleAnalysisOutputsInfo(projectSampleAnalysisOutputInfos);
+	}
+
+	/**
+	 * Get all the automated single sample analysis outputs for the project
+	 *
+	 * @param principal Currently logged in user.
+	 * @return a list of filtered {@link ProjectSampleAnalysisOutputInfo} single sample analysis outputs for the user
+	 */
+	public List<ProjectSampleAnalysisOutputInfo> getUserSingleSampleOutputs(Principal principal) {
+		User user = userService.getUserByUsername(principal.getName());
```

Comment:
As I said above, you can get the user using `SecurityContextHolder`, and therefore don't really need to pass it into here.

Reply:
Updated in c6f2f6cfa2f5b74494bf05a58d24fd7fb08c35fb